### ISSUE TITLE
nn.Transformer.generate_square_subsequent_mask does not behave as expected

### ISSUE THREAD

**Issue by NestLakerJasonLIN** [pytorch/pytorch#29820]
## 🐛 Bug

To generate mask, the `nn.Transformer.generate_square_subsequent_mask` need set all masked positions as `-inf` and all unmasked positions as `0.0`. To do this, the call `masked_fill` set positions in `mask` where `mask == 0` as `-inf`. However, the `mask` is of type `boolean` instead of `float`, so the `mask==0` does not behave as expected and hence result is wrong.
I think we should use `mask.float()==0` instead.

## To Reproduce

Details are shown in the below code snippet.

```
import torch.nn as nn

model = nn.Transformer()
sz = 3
original_mask = model.generate_square_subsequent_mask(sz = sz)
print(original_mask)
```
output:
```
tensor([[0., 0., 0.],
        [-inf, 0., 0.],
        [-inf, -inf, 0.]])
```

## Expected behavior

```
tensor([[0., -inf, -inf],
        [0., 0., -inf],
        [0., 0., 0.]])
```

## Environment

PyTorch version: 0.4.0
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.13.4
GCC version: Could not collect
CMake version: version 3.13.2

Python version: 3.6
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] numpy==1.14.5
[pip3] numpydoc==0.8.0
[pip3] torch==1.3.0.post2
[pip3] torchtext==0.4.0
[pip3] torchvision==0.4.1.post2
[conda] blas                      1.0                         mkl  
[conda] mkl                       2018.0.0             h5ef208c_6  
[conda] mkl-service               1.1.2            py36h7ea6df4_4  
[conda] pytorch                   0.4.0           py36_cuda0.0_cudnn0.0_1    pytorch
[conda] torchvision               0.2.1                    py36_1    pytorch

 - Any other relevant information:
I produce this bug using Jupyter Notebook 4.4.0 with torch==1.3.0.post2

**Comment by ailzhang** [comment]
cc: @zhangguanheng66 might be interested in taking a look?

**Comment by zhangguanheng66** [comment]
It's an old issue related to type promotion and has been fixed by v.1.3.1. Please update your pytorch with the latest binary package or master branch. Feel free to re-open the issue if you still have questions.

**Comment by NestLakerJasonLIN** [comment]
Thanks! It works now after updating.
